### PR TITLE
updatedpodfailurepolicy doc

### LINKED ISSUE #44952: Add `immutable` on `PodFailurePolicyRule` in docs
**This is a Bug Report**

**Problem:**

When applying a `PodFailurePolicy` on an already made `CronJob`, it returns a invalid value stating that the `PodFailurePolicy` is an immutable field. This isn't brought up anywhere when checking the docs.

```
 Invalid value: batch.PodFailurePolicy{Rules:[]batch.PodFailurePolicyRule(nil)}: field is immutable
 ```

**Proposed Solution:**
Update docs so that `PodFailurePolicyRule` is an immutable field in CronJobs

**Page to Update:**
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.29/#podfailurepolicyrule-v1-batch

### PR DESCRIPTION
**Pull request by arujjval**
resolves #44952

### CODE REVIEW

**Comment by stmcginnis** [comment]
Did you intend to include the package-lock.json file? That looks like it would be better in a separate dependency update patch if needed.

File: static/docs/reference/generated/kubernetes-api/v1.29/index.html
Comment (dipesh-rawat) [inline static/docs/reference/generated/kubernetes-api/v1.29/index.html:1]: This page is auto-generated from the upstream source code, any fixes would need to be addressed there. Please refer to the upstream contribution guidelines outlined at: https://kubernetes.io/docs/contribute/generate-ref-docs/contribute-upstream/, https://kubernetes.io/docs/contribute/generate-ref-docs/kubernetes-api/

**Comment by arujjval** [comment]
> Did you intend to include the package-lock.json file? That looks like it would be better in a separate dependency update patch if needed.

I intended to but you are right, it is better for separate patch.

File: static/docs/reference/generated/kubernetes-api/v1.29/index.html
Comment (arujjval) [reply static/docs/reference/generated/kubernetes-api/v1.29/index.html:1]: so do I have to remove the code I added?

**Pull request closed without merging**